CellFabric/Cell_Fabric_FinFET__Mock/fabric_CMB5_NMOS.py

```diff
@@ -0,0 +1,226 @@
+import sys
+import json
+import argparse
+from os import system
+import gen_gds_json
+import gen_lef
+import itertools
+
+from cell_fabric import Via, Region, Canvas, Wire, Pdk
+from cell_fabric import CenterLineGrid, UncoloredCenterLineGrid, ColoredCenterLineGrid
+from cell_fabric import Grid, EnclosureGrid, SingleGrid, CenteredGrid
+
+
+class CanvasNMOS(Canvas):
+
+    
+    def __init__( self, fin_u, fin, finDummy, gate, gateDummy):
+        super().__init__()
+        p = Pdk().load('./../../PDK_Abstraction/FinFET14nm_Mock_PDK/FinFET_Mock_PDK_Abstraction.json')                                          
+        assert   p['Feol']['v0Pitch'] < 2*p['M2']['Pitch']
+
+######### Derived Parameters ############        
+        self.gatesPerUnitCell = gate + 2*gateDummy
+        self.finsPerUnitCell = fin + 2*finDummy
+        # Must be a multiple of 2
+        assert self.finsPerUnitCell % 2 == 0
+        assert finDummy % 2 == 0
+        assert gateDummy > 0
+        # Should be a multiple of 4 for maximum utilization
+        assert self.finsPerUnitCell % 4 == 0
+        self.m2PerUnitCell = self.finsPerUnitCell//2 + 0
+        self.unitCellHeight = self.m2PerUnitCell* p['M2']['Pitch']
+        unitCellLength = self.gatesPerUnitCell* p['Poly']['Pitch']
+        dcPitch  = 2* p['M1']['Pitch']               
+        pcPitch  = self.unitCellHeight//2
+        activeWidth1 =  p['Fin']['Pitch']*fin_u
+        activeWidth =  p['Fin']['Pitch']*fin
+        activeWidth_h = ((gate-1)* p['Poly']['Pitch']) + (p['Feol']['plActive_s']*2)+p['Poly']['Width']
+        activeOffset = activeWidth//2 + finDummy* p['Fin']['Pitch']-p['M2']['Pitch']//2+p['Fin']['Offset']
+        activePitch = self.unitCellHeight
+        RVTPitch = activePitch
+        RVTWidth = activeWidth + 2*p['Feol']['active_enclosure']
+        RVTOffset = RVTWidth//2 + finDummy* p['Fin']['Pitch']-p['Fin']['Pitch']//2-p['Feol']['active_enclosure']+p['Fin']['Offset']
+
+############Include these all ###########
+       # self.extension_x = (self. self.pitch('Poly', 0, p) - self. p['Poly']['Width'])//2  ### Minimum horizontal extension of GCUT past GATE
+
+        self.m0 = self.addGen( Wire( 'm0', 'M0', 'v',
+                                     clg=UncoloredCenterLineGrid( pitch=   p['Feol']['m0Pitch'], width= p['Feol']['m0Width'], offset= p['Feol']['m0Pitch']//2),
+                                     spg=EnclosureGrid( pitch=activePitch, offset=activeOffset, stoppoint=activeWidth//2, check=True)))
+
+        self.m1 = self.addGen( Wire( 'm1', 'M1', 'v',
+                                     clg=ColoredCenterLineGrid( colors=['c1','c2'], pitch= p['M1']['Pitch'], width=p['M1']['Width'], offset= p['M1']['Offset']),
+                                     spg=EnclosureGrid( pitch= p['M2']['Pitch'], offset= p['M2']['Offset'], stoppoint= p['M2']['Width']//2 +p['V1']['VencA_L'], check=True)))
+
+        self.m2 = self.addGen( Wire( 'm2', 'M2', 'h',
```

Comment:
See previous comment about managing BEOL metal layers yourself instead of using DefaultCanvas. VIAs may need additional processing.

Reply:
Yes, I think we require different default BEOL for transistors and Cap/Resistors. As metal layer abstraction is different for them.

Reply:
Would like to understand this comment further. The Mock PDK JSON parameters apply to transistor routing but not for Cap / Resistors?

Reply:
For capacitors we are using minimum spacing for a higher density; I have added a "Cap" layer in the JSON for this spacing abstraction.

Reply:
That's good for now. Please add some tests so that we can refactor later. We can merge once you have removed the redundant CanvasNMOS & optionally based it off of DefaultCanvas. Marking this as resolved.